deltacat/compute/compactor/steps/materialize.py

```diff
@@ -41,6 +41,54 @@ def materialize(
         schema: Optional[pa.Schema] = None,
         deltacat_storage=unimplemented_deltacat_storage) -> MaterializeResult:
 
+    def _materialize(
+            compacted_tables: List[pa.Table],
+            compacted_tables_record_count: int) -> MaterializeResult:
+        compacted_tables_size = sum([TABLE_CLASS_TO_SIZE_FUNC[type(tbl)](tbl)
+                                     for tbl in compacted_tables])
+        logger.debug(f"Uploading {len(compacted_tables)} compacted tables "
+                     f"with size: {compacted_tables_size} bytes "
+                     f"and record count: {compacted_tables_record_count}")
+        compacted_table = pa.concat_tables(compacted_tables)
+        if compacted_file_content_type in DELIMITED_TEXT_CONTENT_TYPES:
+            # convert to pandas since pyarrow doesn't support custom delimiters
+            # and doesn't support utf-8 conversion of all types (e.g. Decimal128)
+            # TODO (pdames): compare performance to pandas-native materialize path
+            df = compacted_table.to_pandas(
+                split_blocks=True,
+                self_destruct=True,
+            )
+            del compacted_table
+            compacted_table = df
+        delta = deltacat_storage.stage_delta(
+            compacted_table,
+            partition,
+            max_records_per_entry=max_records_per_output_file,
+            content_type=compacted_file_content_type,
+        )
+        manifest = delta.manifest
+        manifest_records = manifest.meta.record_count
+        assert(manifest_records == len(compacted_table),
+               f"Unexpected Error: Materialized delta manifest record count "
+               f"({manifest_records}) does not equal compacted table record count "
+               f"({len(compacted_table)})")
+
+        materialize_result = MaterializeResult.of(
+            delta,
+            mat_bucket_index,
+            # TODO (pdames): Generalize WriteResult to contain in-memory-table-type
+            #  and in-memory-table-bytes instead of tight coupling to paBytes
+            PyArrowWriteResult.of(
+                len(manifest.entries),
+                TABLE_CLASS_TO_SIZE_FUNC[type(compacted_table)](compacted_table),
+                manifest.meta.content_length,
+                len(compacted_table),
+            ),
+        )
+        del compacted_table
```

Comment:
I assume del would not be needed here as this goes out of scope anyway.

Reply:
Yeah, the intention here was to make it more explicit that we want to to free this table from memory.

Reply:
redundant code always causes confusion. The next reader here would be thinking why do we have to del here, and are there implications of removing this.

Reply:
Agreed, we could probably get rid of all `del` usages in this function altogether.